### ISSUE TITLE
error unions + optionals generating invalid llvm

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#3692]
```zig
const std = @import("std");

test "catch default null" {
    const should_be_null: ?*Foo = bar() catch null;
    std.testing.expect(should_be_null == null);
}

const Foo = struct {
    field: i32,
};

fn bar() anyerror!*Foo {
    return error.NotImplemented;
}
```

expected to pass. instead:

```
[nix-shell:~/Downloads/zig/build]$ ./zig test test.zig 
broken LLVM module found: Instruction does not dominate all uses!
  %7 = load %Foo*, %Foo** %6, align 8, !dbg !1351
  store %Foo* %7, %Foo** %should_be_null, align 8, !dbg !1352
```

**Comment by andrewrk** [comment]
Fixed by #4404

**Comment by andrewrk** [comment]
Landed in 014f66e6de4aaf81f32c796b12f981326a479397
